apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -0,0 +1,446 @@
+
+  const deviceName = useMemo(
+    () => getDeviceModel(device.modelId).productName,
+    [device.modelId],
+  );
+
+  const defaultSteps: UpdateSteps = useMemo(
+    () => ({
+      prepareUpdate: {
+        status: VerticalStepper.ItemStatus.inactive,
+        title: t("FirmwareUpdate.steps.prepareUpdate.titleActive"),
+        renderBody: () => (
+          <Text color="neutral.c80">
+            {t("FirmwareUpdate.steps.prepareUpdate.description", {
+              deviceName,
+            })}
+          </Text>
+        ),
+      },
+      installUpdate: {
+        status: VerticalStepper.ItemStatus.inactive,
+        title: t("FirmwareUpdate.steps.installUpdate.titleInactive"),
+        renderBody: () => (
+          <Text color="neutral.c80">
+            {t("FirmwareUpdate.steps.installUpdate.description", {
+              deviceName,
+            })}
+          </Text>
+        ),
+      },
+      restoreAppsAndSettings: {
+        status: VerticalStepper.ItemStatus.inactive,
+        title: t("FirmwareUpdate.steps.restoreSettings.titleInactive"),
+        renderBody: () => (
+          <Flex>
+            <Text color="neutral.c80">
+              {t("FirmwareUpdate.steps.restoreSettings.description")}
+            </Text>
+            {/* TODO: create custom component here with its own state for the restoring */}
+            <VerticalStepper nested steps={restoreSteps} />
+          </Flex>
+        ),
+      },
+    }),
+    [t, deviceName],
+  );
+
+  useEffect(() => {
+    dispatch(updateMainNavigatorVisibility(false));
+    return () => {
+      // this works because the firmware update screen has only one way out which is going back
+      // and it will unmount the screen and put the visibility back on
+      // if there other navigations for the firmware update in the future we might need to adapt this
+      dispatch(updateMainNavigatorVisibility(true));
+    };
+  }, [dispatch]);
+
+  const [isCloseWarningOpen, setIsCloseWarningOpen] = useState(false);
+
+  const isAllowedToClose = useMemo(() => {
+    const closableSteps: UpdateFirmwareActionState["step"][] = [
+      "firmwareUpdateCompleted",
+      "preparingUpdate",
+      "installOsuDevicePermissionDenied",
+    ];
+
+    return closableSteps.includes(updateState.step);
+  }, [updateState.step]);
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerRight: () => (
+        <Button
+          onPress={() => {
+            if (isAllowedToClose) {
+              quitUpdate();
+            } else {
+              setIsCloseWarningOpen(true);
+            }
+          }}
+          Icon={Icons.CloseMedium}
+        />
+      ),
+    });
+  }, [navigation, quitUpdate, isAllowedToClose]);
+
+  // this will depend on the steps we go through during the update
+  const [totalNumberOfSteps, setTotalNumberOfSteps] = useState(2);
+
+  useEffect(() => {
+    // TODO: instead of time out, trigger when all the backups have been made
+    triggerUpdate();
```

Comment:
No timeout nor waiting for backup for now ;)

Reply:
Good catch, removed the comment